pkgs/tools/filesystems/wipefreespace/default.nix

```diff
@@ -0,0 +1,45 @@
+{ stdenv
+, lib
+, fetchurl
+, e2fsprogs
+, ntfs3g
+, xfsprogs
+, reiserfsprogs
+, reiser4progs
+, jfsutils
+, libuuid
+, texinfo
+}:
+
+stdenv.mkDerivation rec {
+  pname = "wipefreespace";
+  version = "2.5";
+
+  src = fetchurl {
+    url = "mirror://sourceforge/project/wipefreespace/wipefreespace/${version}/wipefreespace-${version}.tar.gz";
+    sha256 = "sha256-wymV6G4Et5TCoIztZfdb3xuzjdBHFyB5OmI4EcsJKwQ=";
+  };
+
+  # missed: FAT12/16/32 MinixFS HFS+ OCFS
+  buildInputs = [
+    e2fsprogs
+    ntfs3g
+    xfsprogs
+    reiserfsprogs
+    reiser4progs
+    jfsutils
```

Comment:
These don't seem to be working yet.

```console
checking for uuid_unparse in -luuid... yes
checking for misc_test_bit in -lmisc... no
checking for reiserfs_open in -lcore... no
checking for reiser3_new_bread... no
checking for aal_device_open in -laal... no
configure: WARNING: Reiser4 support impossible without the aal and uuid libraries.
[...]
checking for ujfs_get_superblk in -lfs... no

[...]

 *      ReiserFSv3: no (library missing or unusable)
 *      Reiser4: no (library missing or unusable)
 [...]
 *      JFS: no (library missing or unusable)
 ```

Reply:
https://wipefreespace.sourceforge.io/docs/Filesystems-support.html

Reiser3 needs modifications to what `reiserfsprogs` installs and might complicate configuration abit, maybe not worth it.
> Support: using the library from `reiserfsprogs`, but that library is not a part of any official development package.
> [...]
> - the development package for reiserfsprogs (header files - *.h - and compiled static libraries - *.a - from the package). 
> 
> [...]
> **WARNING**: If both ReiserFSv3 and ReiserFSv4 are enabled, the program may refuse to link
> **WARNING**: If both ReiserFSv3 and MinixFS are enabled, the program may refuse to link
> **WARNING**: If both ReiserFSv3 and JFS are enabled, you **MUST** either disable the support for either of these filesystems (you can compile two versions of the program) or fix the ReiserFSv3 library yourself

Reiser4 is just missing `libaal`.
> - development package for `reiser4progs`, `libuuid` (`e2fsprogs` or `e2fsprogs-libs`) and `libaal`

JFS also needs modifications to what `jfsutils` installs.
> Compile `jfsutils`, then copy all the `jfs_*.h` header files from the `include` subdirectory and the `libfs.a` file to a directory where the C compiler can find them

Reply:
@OPNA2608 thanks. To be honest hacking `jfsutils` seems as the wrong way. And the way to exclude support of Reiser3 is also seems like a easy win.

Reply:
@OPNA2608 thanks; I've fixed reiser4 and JFS; removed attempt to use reiser3.

Right now it is configured like:
```
 *      Ext2/3/4: yes
 *      NTFS: yes
 *      XFS: yes
 *      ReiserFSv3: no (library missing or unusable)
 *      Reiser4: yes
 *      FAT12/16/32: no (library missing or unusable)
 *      MinixFS: no (library missing or unusable)
 *      JFS: yes
 *      HFS+: no (library missing or unusable)
 *      OCFS: no (library missing or unusable)
```